app/collections/home.cjsx

```diff
@@ -0,0 +1,45 @@
+React = require 'react'
+CollectionCreateForm = require './create-form'
+talkClient = require '../api/talk'
+apiClient = require '../api/client'
+authClient = require '../api/auth'
+auth = require '../api/auth'
+{Link} = require 'react-router'
+ChangeListener = require '../components/change-listener'
+PromiseRenderer = require '../components/promise-renderer'
+
+module?.exports = React.createClass
+  displayName: 'CollectionsHome'
+
+  getInitialState: ->
+    collections: []
+
+  componentWillMount: ->
+    @setCollections()
+
+  setCollections: ->
+    auth.checkCurrent().then (user) =>
+      user.get('collections')
+        .then (collections) =>
+          @setState {collections}
+        .catch (e) -> throw new Error(e)
+
+  collectionLink: (d, i) ->
+    <div key={i}>
```

Comment:
React uses the `key` of components in an array to associate them with their data. So if you have a list of three objects `[a, b, c]` and pass them in props to three components keyed by index, and then shift off `a` from the array, the third component (which represented `c`) will be removed (because index `2` no longer exists) and you'll be left with two components. The first will maintain the _state_ it had when it represented `a`, but it'll now have `b` in its _props_. The second will have its state from `b` but with `c` in its props.

Where's React use it in their examples? That seems weird.